src/wallet/rpcwallet.cpp

```diff
@@ -2036,7 +2036,7 @@ UniValue walletpassphrase(const UniValue& params, bool fHelp)
         anonymizeOnly = params[2].get_bool();
 
     if (!pwalletMain->IsLocked() && pwalletMain->fWalletUnlockAnonymizeOnly && anonymizeOnly)
-        throw JSONRPCError(RPC_WALLET_ALREADY_UNLOCKED, "Error: Wallet is already unlocked.");
+        throw JSONRPCError(RPC_WALLET_ALREADY_UNLOCKED, "Error: Wallet is already unlocked for anonymization only.");
```

Comment:
I was fixing message here to match conditions in `if` but I guess you are right, we should probably have more `RPC_WALLET_ALREADY_UNLOCKED` validations here. Not sure about `7` though, I can think of 1 only so far:
```
if (!pwalletMain->IsLocked() && !pwalletMain->fWalletUnlockAnonymizeOnly && !anonymizeOnly)`
throw JSONRPCError(RPC_WALLET_ALREADY_UNLOCKED, "Error: Wallet is already fully unlocked.");
```
cases where `pwalletMain->fWalletUnlockAnonymizeOnly != anonymizeOnly` are fine imo, it's simply means that one is changing unlock mode from full to mixing and vice versa. Such situations are normal and should be handled properly by wallet Unlock() function (now).